### PR TITLE
Applies patch contributed by @nashhoogwater

### LINKED ISSUE #35: NPE when running plugin via Kitchen environment
See forum: https://datacleaner.org/topic/1122/Help-with-Pentho-Kitchen-running-job-file-with-Execute-DataCleaner-job

Even includes a suggestion for a fix :)

### PR DESCRIPTION
**Pull request by LosD**
Fixes NPE in Kitchen, as well as variable substitution.

Fixes #35

### CODE REVIEW

File: src/main/java/org/pentaho/di/profiling/datacleaner/ModelerHelper.java [hunk 1]
```diff
@@ -405,7 +408,7 @@ public void profileStep(final boolean buildJob) throws Exception {
                 final JaxbJobWriter jobWriter = new JaxbJobWriter(dataCleanerConfiguration);
                 jobWriter.write(analysisJob, jobOutputStream);
             } catch (Exception e) {
-                final LogChannelInterface log = Spoon.getInstance().getLog();
+				LogChannelInterface log = new LogChannel( LOGCHANNEL_NAME );
```
Comment (ClaudiaPHI) [inline src/main/java/org/pentaho/di/profiling/datacleaner/ModelerHelper.java]: final, indentation

File: src/main/java/org/pentaho/di/profiling/datacleaner/ModelerHelper.java [hunk 2]
```diff
@@ -136,7 +139,7 @@ public static int launchDataCleaner(DataCleanerSpoonConfiguration dataCleanerSpo
             String jobFile, String datastore, String dataFile, String outputFiletype, String outputFilename,
             String additionalArguments, boolean profileStep) {
 
-        final LogChannelInterface log = Spoon.getInstance().getLog();
+		LogChannelInterface log = new LogChannel( LOGCHANNEL_NAME );
         int exitCode = 0;
```
Comment (ClaudiaPHI) [inline src/main/java/org/pentaho/di/profiling/datacleaner/ModelerHelper.java:143]: final, indentation

**Comment by ClaudiaPHI** [comment]
LGTM!

File: src/main/java/org/pentaho/di/profiling/datacleaner/ModelerHelper.java (hunk 2 above)
Comment (LosD) [inline src/main/java/org/pentaho/di/profiling/datacleaner/ModelerHelper.java:143]: Heh. It might have been a plan if I reviewed the patch I merged!

**Pull request merged**